node/node.go

```diff
@@ -114,6 +116,47 @@ func NewNode(reg *prometheus.Registry, config *Config, pubIPProvider pubip.Provi
 	// Create ChainState Client
 	cst := eth.NewChainState(tx, client)
 
+	var keyPair *core.KeyPair
+	var blsClient blssignerV1.SignerClient
+	if config.PrivateBls != "" {
+		nodeLogger.Info("using local keystore private key for BLS signing")
+		// Generate BLS keys
+		keyPair, err = core.MakeKeyPairFromString(config.PrivateBls)
+		if err != nil {
+			return nil, err
+		}
+
+		config.ID = keyPair.GetPubKeyG1().GetOperatorID()
+	} else {
+		pkBytes, err := hex.DecodeString(config.BLSPublicKeyHex)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode BLS public key: %w", err)
+		}
+		pubkey := new(core.G1Point)
+		publicKey, err := pubkey.Deserialize(pkBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		config.ID = publicKey.GetOperatorID()
+
+		nodeLogger.Info("creating signer client", "url", config.BLSRemoteSignerUrl)
+		creds := insecure.NewCredentials()
+		if config.BLSSignerTLSCertFilePath != "" {
+			creds, err = credentials.NewClientTLSFromFile(config.BLSSignerTLSCertFilePath, "")
+			if err != nil {
+				return nil, err
+			}
+		}
+		conn, err := grpc.NewClient(
```

Comment:
What happens if this connection gets broken while node is running?

Reply:
A grpc « connection » is really a virtual connection. It’s not 1-1 with the underlying tcp connection. If tcp conn drops the grpc runtime dials a new connection.

don’t think grpc « connection » can ever be disconnected. But worth checking if there are conditions that need to be checked/handled.